src/lib/pages/my-module-verifications/components/MoveVerifyTaskStatusFilter.tsx

```diff
@@ -0,0 +1,136 @@
+  { label: "completed", value: MoveVerifyTaskStatus.Finished },
+  { label: "failed", value: MoveVerifyTaskStatus.NotFound },
+  { label: "pending", value: MoveVerifyTaskStatus.Pending },
+  { label: "verifying", value: MoveVerifyTaskStatus.Running },
+];
+
+export const MoveVerifyTaskStatusFilter = forwardRef<
+  HTMLInputElement,
+  MoveVerifyTaskStatusFilterProps
+>(
+  (
+    {
+      result,
+      minW = "50%",
+      setResult,
+      placeholder,
+      label,
+      isMulti,
+    }: MoveVerifyTaskStatusFilterProps,
+    ref
+  ) => {
+    const [keyword, setKeyword] = useState("");
+    const [isDropdown, setIsDropdown] = useState(false);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const boxRef = useRef<HTMLDivElement>(null);
+
+    const dropdownValue = useMemo(
+      () =>
+        keyword
+          ? matchSorter(OPTIONS, keyword, {
+              keys: ["label"],
+              threshold: matchSorter.rankings.CONTAINS,
+            })
+          : OPTIONS,
+      [keyword]
+    );
+
+    const isOptionSelected = (option: MoveVerifyTaskStatus) =>
+      result.some((selectedOption) => selectedOption === option);
+
+    const selectOption = (option: MoveVerifyTaskStatus) => {
+      setKeyword("");
+
+      if (!isMulti) {
+        setIsDropdown(false);
+
+        if (result[0] === option) setResult([]);
+        else setResult([option]);
+      } else {
+        setResult(toggleItem(result, option));
+      }
+    };
+
+    useOutsideClick({
+      ref: boxRef,
+      handler: () => setIsDropdown(false),
+    });
+
+    return (
+      <FormControl ref={boxRef} minW={minW} maxW="full" h={8}>
+        <FilterInput
+          keyword={keyword}
+          placeholder={placeholder}
+          result={result}
+          label={label}
+          inputRef={inputRef}
+          ref={ref}
+          isDropdown={isDropdown}
+          setKeyword={setKeyword}
+          setIsDropdown={setIsDropdown}
+          chipContainerComponent={
+            <Flex alignItems="center" pl={2} gap={2}>
+              {result.map((option) => (
+                <FilterChip
+                  key={option}
+                  chipComponent={
+                    <MyModuleVerificationDetailsStatusBadge
+                      status={option}
+                      hasCloseBtn
+                    />
+                  }
+                  onSelect={() => setResult(toggleItem(result, option))}
+                />
+              ))}
+            </Flex>
+          }
+        />
+
+        {isDropdown && (
+          <DropdownContainer>
+            {!dropdownValue.length && <Flex p={2}>No filter matched</Flex>}
+
+            {/* option selection section */}
+            {dropdownValue.map((option) => (
+              <FilterDropdownItem
+                key={option.label}
+                filterDropdownComponent={
+                  <MyModuleVerificationDetailsStatusBadge
+                    status={option.value}
+                  />
```

Comment:
Just wondering, the blue dot with label Verifying still need to animate in status dropdown?